src/main/java/de/tum/cit/aet/artemis/iris/service/settings/IrisSettingsService.java

```diff
@@ -276,9 +300,135 @@ private IrisCourseSettings updateCourseSettings(IrisCourseSettings existingSetti
+        }
+
+        var newEnabledForCategoriesTextExerciseChat = existingSettings.getIrisTextExerciseChatSettings() == null ? new TreeSet<String>()
+                : existingSettings.getIrisTextExerciseChatSettings().getEnabledForCategories();
+        if (!Objects.equals(oldEnabledForCategoriesTextExerciseChat, newEnabledForCategoriesTextExerciseChat)) {
+            textExerciseRepository.findAllWithCategoriesByCourseId(existingSettings.getCourse().getId())
+                    .forEach(exercise -> setEnabledForExerciseByCategories(exercise, oldEnabledForCategoriesTextExerciseChat, newEnabledForCategoriesTextExerciseChat));
+        }
+
         return irisSettingsRepository.save(existingSettings);
     }
 
+    /**
+     * Set the enabled status for an exercise based on it's categories.
+     * Compares the old and new enabled categories, reads the exercise categories,
+     * and updates the Iris chat settings accordingly if the new enabled categories match any of the exercise categories.
+     * This method is used when the enabled categories of the course settings are updated.
+     *
+     * @param exercise                The exercise to update the enabled status for
+     * @param oldEnabledForCategories The old enabled categories
+     * @param newEnabledForCategories The new enabled categories
+     */
+    public void setEnabledForExerciseByCategories(Exercise exercise, SortedSet<String> oldEnabledForCategories, SortedSet<String> newEnabledForCategories) {
+        var removedCategories = new TreeSet<>(oldEnabledForCategories == null ? Set.of() : oldEnabledForCategories);
+        removedCategories.removeAll(newEnabledForCategories);
+        var categories = getCategoryNames(exercise.getCategories());
+
+        if (newEnabledForCategories != null && categories.stream().anyMatch(newEnabledForCategories::contains)) {
+            var exerciseSettings = getRawIrisSettingsFor(exercise);
+            if (exercise instanceof ProgrammingExercise) {
+                exerciseSettings.getIrisChatSettings().setEnabled(true);
+            }
+            else if (exercise instanceof TextExercise) {
+                exerciseSettings.getIrisTextExerciseChatSettings().setEnabled(true);
+            }
+            irisSettingsRepository.save(exerciseSettings);
+        }
+        else if (categories.stream().anyMatch(removedCategories::contains)) {
+            var exerciseSettings = getRawIrisSettingsFor(exercise);
+            if (exercise instanceof ProgrammingExercise) {
+                exerciseSettings.getIrisChatSettings().setEnabled(false);
+            }
+            else if (exercise instanceof TextExercise) {
+                exerciseSettings.getIrisTextExerciseChatSettings().setEnabled(false);
+            }
+            irisSettingsRepository.save(exerciseSettings);
+        }
```

Comment:
The body of the outer if-else-if is really similar. How about a helper method? Especially since this is also repeated in the next method